### PR TITLE
Added Subcategory to API

### PR DESCRIPTION
**Pull request by giosce**

### CODE REVIEW

File: tools/api-server/server.js [hunk 1]
```diff
@@ -79,49 +79,70 @@ app.get('/topics_projects', cors(), function (req, res) {
...
 		  }
-		  if(it != row.display_name) {
-			it = row.display_name;
+		  if(row.subcategory != null && subcat != row.subcategory) {
+			  subcat = row.subcategory;
+			  //console.log("New subcategory: " + subcat);
+			  subcateg = new Object();
+			  subcateg.text = subcat;
+			  subcategories[sc] = categ;
+			  subcateg.children = new Array();
```
Comment (k3KAW8Pnf7mkmdSMPHz27) [inline tools/api-server/server.js:124]: Unless it changes the logic, perhaps simplify to 

```suggestion
			  subcateg = { text: subcat, children: [] };
			  subcategories[sc] = categ;
```

and remove `subcategories` if it isn't used

File: tools/api-server/server.js [hunk 2]
```diff
@@ -79,49 +79,70 @@ app.get('/topics_projects', cors(), function (req, res) {
...
 
 	  var cat = "";
+	  var subcat = "";
+	  var subcateg = null;
 	  var categories = new Array();
+	  var subcategories = new Array();
 	  var c = 0;
-	  var it = "";
+	  var sc = 0;
+	  var item_name = "";
```
Comment (k3KAW8Pnf7mkmdSMPHz27) [inline tools/api-server/server.js:103]: `it` was not very meaningful, thank you :)

**Review by k3KAW8Pnf7mkmdSMPHz27 — COMMENTED** [review]
I haven't tried running it locally, but perhaps delete `subcategories` and it looks good.

If you don't mind that I merge it, I'll have time to run it locally tomorrow and just make sure that nothing was missed.

**Comment by giosce** [comment]
I believe I have addressed Jonatan code review findings.
And the html file changes also made in this PR.

**Review by k3KAW8Pnf7mkmdSMPHz27 — APPROVED** [review]
Still can't run it locally, but I took another sweep over the changes and they look good to me! (and all my code review comments were addressed)

**Pull request merged**